keys.go

```diff
@@ -133,103 +133,96 @@ type KeyVersion struct {
-	key := c.createKeyTemplate(ctx, name, expiration, payload, encryptedNonce, iv, extractable, aliases, AlgorithmRSAOAEP1, nil)
-	return c.createKey(ctx, key)
-}
-
-// CreateRootKey creates a new, non-extractable key resource without
-// key material.
-func (c *Client) CreateRootKey(ctx context.Context, name string, expiration *time.Time) (*Key, error) {
-	return c.CreateKey(ctx, name, expiration, false)
+func WithExpiration(expiration *time.Time) CreateKeyOption {
+	return func(key *Key) string {
+		key.Expiration = expiration
+		return "expiration"
+	}
 }
 
-// CreateStandardKey creates a new, extractable key resource without
-// key material.
-func (c *Client) CreateStandardKey(ctx context.Context, name string, expiration *time.Time) (*Key, error) {
-	return c.CreateKey(ctx, name, expiration, true)
+func WithDescription(description string) CreateKeyOption {
+	return func(key *Key) string {
+		key.Description = description
+		return "description"
+	}
 }
 
-// CreateImportedRootKey creates a new, non-extractable key resource
-// with the given key material.
-func (c *Client) CreateImportedRootKey(ctx context.Context, name string, expiration *time.Time, payload, encryptedNonce, iv string) (*Key, error) {
-	return c.CreateImportedKey(ctx, name, expiration, payload, encryptedNonce, iv, false)
+func WithPayload(payload, encryptedNonce, iv string, sha1 bool) CreateKeyOption {
+	return func(key *Key) string {
+		ret := "payload"
+		key.Payload = payload
+		if !key.Extractable {
+			algorithm := AlgorithmRSAOAEP256
+			if sha1 {
+				algorithm = AlgorithmRSAOAEP1
+			}
+			key.EncryptedNonce = encryptedNonce
+			key.IV = iv
+			key.EncryptionAlgorithm = algorithm
+			ret += "-" + algorithm
+		}
+		return ret
+	}
 }
 
-// CreateStandardKey creates a new, extractable key resource with the
-// given key material.
-func (c *Client) CreateImportedStandardKey(ctx context.Context, name string, expiration *time.Time, payload string) (*Key, error) {
-	return c.CreateImportedKey(ctx, name, expiration, payload, "", "", true)
+func WithAliases(aliases []string) CreateKeyOption {
+	return func(key *Key) string {
+		key.Aliases = aliases
+		return "aliases"
+	}
 }
 
-// CreateKeyWithAliaes creats a new key with alias names. A key can have a maximum of 5 alias names.
-// For more information please refer to the links below:
-// https://cloud.ibm.com/docs/key-protect?topic=key-protect-create-root-keys#create-root-key-api
-// https://cloud.ibm.com/docs/key-protect?topic=key-protect-create-standard-keys#create-standard-key-api
-func (c *Client) CreateKeyWithAliases(ctx context.Context, name string, expiration *time.Time, extractable bool, aliases []string) (*Key, error) {
-	return c.CreateImportedKeyWithAliases(ctx, name, expiration, "", "", "", extractable, aliases)
+func WithPolicy(policy *Policy) CreateKeyOption {
+	return func(key *Key) string {
+		key.Rotation = policy.Rotation
+		key.DualAuthDelete = policy.DualAuth
+		return "policy"
```

Comment:
I needed these functions to return a string to ensure i knew when a policy was passed, so that it would call the `create key with policy overrides function`, even when a policy with empty/nil fields is passed.